src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -468,6 +453,7 @@ internal async Task Open(NpgsqlTimeout timeout, bool async, CancellationToken ca
                 using (StartCancellableOperation(cancellationToken, attemptPgCancellation: false))
                 {
                     await Authenticate(username, timeout, async, cancellationToken);
+                    canRetry = false;
```

Comment:
Won't this retry on various other non-related errors? For example, say I connect with Prefer but then get a simple connection refused error (PostgreSQL is down) - we shouldn't retry in that scenario, right?